include/boost/geometry/strategies/cartesian/turn_in_ring_winding.hpp

```diff
@@ -0,0 +1,211 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2020 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_CARTESIAN_TURN_IN_RING_WINDING_HPP
+#define BOOST_GEOMETRY_STRATEGIES_CARTESIAN_TURN_IN_RING_WINDING_HPP
+
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/config.hpp>
+#include <boost/geometry/algorithms/detail/make/make.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace buffer
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+
+enum place_on_ring_type
+{
+    // +----offsetted----> (offsetted is considered as outside)
+    // |                 |
+    // |                 |
+    // left              right (first point outside, rest inside)
+    // |                 |
+    // |                 |
+    // <-----original----+ (original is considered as inside)
+    place_on_ring_offsetted,
+    place_on_ring_original,
+    place_on_ring_to_offsetted,
+    place_on_ring_from_offsetted,
+};
+
+template <typename CalculationType>
+class turn_in_ring_winding
+{
+
+    // Implements the winding rule.
+    // Basic calculations (on a clockwise ring of 5 segments)
+    // (as everywhere in BG, -1 = right, 0 = on segment, +1 = left)
+    // +--------2--------+  // P : For 1/3, nothing happens, it returns
+    // |                 |  //     For 2, side is right (-1), multiplier=2, -2
+    // |        P        |  //     For 4, side is right (-1), multiplier=1, -1
+    // 1                 3  //     For 5, side is right (-1), multiplier=1, -1, total -4
+    // |             Q   |  // Q : For 2: -2, for 4: -2, total -4
+    // |                 |  // R : For 2: -2, for 5: +2, total 0
+    // +----5---*----4---+  // S : For 2: -1, 3: nothing, 4: +1, total 0
+    //
+    //     R             S
+    //
+
+
+public:
+
+    struct counter
+    {
+        inline counter()
+            : m_count(0)
+            , m_min_distance((std::numeric_limits<CalculationType>::max)())
```

Comment:
Can `CalculationType` be user-defined? For user defined types max will be initialized by `T()`. Can this be a problem?

Reply:
Isn't this a problem then everywhere where we use `numeric_limits`?
But anyway, here it was easy to get rid of it, done
:heavy_check_mark:

Reply:
> Isn't this a problem then everywhere where we use `numeric_limits`?
Yes.